library/std/src/sys_common/rt.rs

```diff
@@ -0,0 +1,62 @@
+use crate::sync::Once;
+use crate::sys;
+use crate::sys_common::thread_info;
+use crate::thread::Thread;
+
+// One-time runtime initialization.
+// Runs before `main`.
+#[cfg_attr(test, allow(dead_code))]
+pub fn init(argc: isize, argv: *const *const u8) {
+    static INIT: Once = Once::new();
+    INIT.call_once(|| unsafe {
+        // SAFETY: Only called once during runtime initialization.
+        sys::init(argc, argv);
+
+        let main_guard = sys::thread::guard::init();
+        // Next, set up the current Thread with the guard information we just
+        // created. Note that this isn't necessary in general for new threads,
+        // but we just do this to name the main thread and to give it correct
+        // info about the stack bounds.
+        let thread = Thread::new(Some("main".to_owned()));
+        thread_info::set(main_guard, thread);
+    });
+}
+
+// One-time runtime cleanup.
+// Runs after `main` or at program exit. Note however that this is not guaranteed to run,
+// for example when the program aborts.
+#[cfg_attr(test, allow(dead_code))]
+pub fn cleanup() {
+    static CLEANUP: Once = Once::new();
+    CLEANUP.call_once(|| unsafe {
+        // SAFETY: Only called once during runtime cleanup.
+        sys::cleanup();
+        // Flush stdout and disable buffering.
+        crate::io::cleanup();
```

Comment:
It might be good to swap these calls. Then if flushing the io needs anything from `sys`, it's still available. (Which might become relevant in the future if we ever add something like custom output redirecting for example.)